include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -804,7 +805,8 @@ __find_subrange(_RandomAccessIterator __first, _RandomAccessIterator __last, _Ra
         return __last;
     }
 
-    auto __unary_pred = __equal_value_by_pred<_Tp, _BinaryPredicate&>(__value, __pred);
+    auto __unary_pred =
+        [__pred, &__value](auto&& __val) mutable { return __pred(std::forward<decltype(__val)>(__val), __value); };
```

Comment:
Why is `__value` captured by reference in this lambda?

Reply:
At least, because `__value `is passed into the function by reference  - `const _Tp& __value`

Reply:
Ah yes, so I guess it is to avoid an unnecessary copy.  What tripped me up is that capturing into a lambda, this looks like a lvalue ref capture, rather than a const lvalue ref capture.  

We could add some extra verbosity to indicate a const ref (`&__value = std::as_const(__value)`), but I think it is not worth it, because I don't think it changes the code functionally.